utils/rate_limiter.go

```diff
@@ -62,8 +65,9 @@ func (b *rateLimiter) CheckCredit(itemCost float64) bool {
 	b.lastTick = currentTime
 	// calculate how much credit have we accumulated since the last tick
 	b.balance += elapsedTime.Seconds() * b.creditsPerSecond
-	if b.balance > b.creditsPerSecond {
-		b.balance = b.creditsPerSecond
+	upperbound := math.Max(b.creditsPerSecond, itemCost)
```

Comment:
just so i dont do anything stupid, you want me to allow the caller to pass in the parameter maxBalance to CheckCredit?